### PR TITLE
feat: trigger events for sample data

### PR DESCRIPTION
**Pull request by abuaboud**
## What does this PR do?

Trigger Events for Sample data such as webhooks.

### CODE REVIEW

File: packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/edit-step-sidebar/test-step/test-step.component.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    this.saveAfterNewDataIsLoaded$ = this.store
+      .select(BuilderSelectors.selectCurrentStep)
+      .pipe(
+        take(1),
+        tap((step) => {
+          if (step && step.type === TriggerType.WEBHOOK) {
+            const clone = { ...step };
+            clone.settings = {
+              inputUiInfo: {
+                historicalData:
```
Comment (abuaboud) [inline packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/edit-step-sidebar/test-step/test-step.component.ts]: Why not query directly from the list trigger events endpoint?

File: packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/edit-step-sidebar/test-step/test-step.component.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                  newResults || clone.settings.inputUiInfo.historicalData,
+                currentSelectedData: this.selectedDataControl.value,
```
Comment (abuaboud) [inline packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/edit-step-sidebar/test-step/test-step.component.ts:152]: maybe `currentSampleData`?

File: packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/step-type-sidebar/step-type-sidebar.component.ts [hunk 3]
```diff
@@ -189,7 +189,9 @@ export class StepTypeSidebarComponent implements OnInit {
           ...base,
           valid: true,
           type: TriggerType.WEBHOOK,
-          settings: {},
+          settings: {
+            inputUiInfo: { currentSelectedData: '', historicalData: [] },
```
Comment (abuaboud) [inline packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/step-type-sidebar/step-type-sidebar.component.ts]: Let's remove historicalData from there

File: packages/shared/src/lib/flows/sample-data.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { Type, Static } from '@sinclair/typebox';
+export const SampleDataSettingsObject =Type.Object({
+    historicalData:Type.Array(Type.Object({
```
Comment (abuaboud) [inline packages/shared/src/lib/flows/sample-data.ts]: Need to be removed

File: packages/frontend/src/app/modules/flow-builder/store/builder/builder.selector.ts [hunk 5]
```diff
@@ -164,6 +165,35 @@ export const selectCurrentStep = createSelector(
...
+const selectCurrentStepSettings = createSelector(
+  selectCurrentStep,
+  (selectedStep) => {
+    if (selectedStep) {
+      return selectedStep.settings;
+    }
+    return undefined;
+  }
+);
+const selectStepHistoricalSampleData = createSelector(
```
Comment (abuaboud) [inline packages/frontend/src/app/modules/flow-builder/store/builder/builder.selector.ts]: I think we can remove this as well.

**Review by AbdulTheActivePiecer — APPROVED** [review]

**Pull request merged**